mapie/classification.py

```diff
@@ -943,15 +952,38 @@ def fit(
 
         estimator = check_estimator_classification(
             X,
-            y_enc,
+            y,
             cv,
             self.estimator
         )
         self.n_features_in_ = check_n_features_in(X, cv, estimator)
 
         n_samples = _num_samples(y)
-        self.n_classes_ = len(np.unique(y))
-        self.classes_ = np.unique(y)
+
+        if self.cv == "prefit":
```

Comment:
Could we not implement this as a function? say: `check_consistency_labels()`?

Reply:
done